src/parser/jobs/brd/modules/RagingStrikes.tsx

```diff
@@ -63,6 +76,73 @@ class BarrageEvaluator extends ExpectedActionsEvaluator {
 	}
 }
 
+interface BuffEvaluatorOptions {
+	actions: ActionKey[]
+	predicate: (action: ActionKey, bestBeforeGCD: number, window: HistoryEntry<EvaluatedAction[]>) => boolean
+
+	suggestionIcon: string
+	severityTiers: SeverityTiers
+}
+
+class BuffEvaluator implements WindowEvaluator, BuffEvaluatorOptions {
+	actions: ActionKey[]
+	predicate: (action: ActionKey, bestBeforeGCD: number, window: HistoryEntry<EvaluatedAction[]>) => boolean
```

Comment:
I think this is a good implementation, and will have value being generalized to core so Dancer (Tech Step/Devilment) and other jobs can use it.  To that end, I would like to see the predicate function be moved into the evaluator as a private logic function so that it can be easily shared.  

If you change actions to be an `Array<Action['id']>` type (requiring implementers to pass in the IDs of the actions being considered, rather than their ActionKey), then you don't wind up with a hidden dependency on Data and the `wasActionUsedInTime` function can be a function of the Evaluator directly instead of having to pass it in from the BuffWindow analyser.